src/utils/hooks.js

```diff
@@ -19,3 +19,68 @@ export function useStateWithStorage(storageKey, initialValue) {
 
 	return [value, setValue];
 }
+
+/**
+ * A custom hook that manages the importance of lists and sorts them accordingly.
+ * @param {Array<Object>} lists - An array of list objects to manage, each containing at least a `path` property.
+ * @returns {Object} An object containing:
+ *  - {Array<Object>} sortedLists - An array of lists sorted by importance and date they were pinned.
+ *  - {Function} setImportantList - A function to toggle the importance of a list identified by its `path`.
+ *  - {Function} isListImportant - A function to check if a given list is marked as important.
+ */
+export function useImportance(lists) {
+	const [sortedLists, setSortedLists] = useState([]);
+	const [importantLists, setImportantLists] = useState(
+		() => JSON.parse(localStorage.getItem('importantLists')) || [],
+	);
+
```

Comment:
use state to display real-time changes to local storage where we store an array of objects:
```
{ 
   path: string,
   datePinned: Date 
}
```
datePinned is needed to be able to display the lists in the reverse add order